src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -297,6 +305,10 @@ public function getName()
             if (ctype_digit($this->name[0])) {
                 $this->name = '_'.$this->name;
             }
+
+            if ('app' !== $this->name) {
+                @trigger_error(sprintf('Auto-discovery of the kernel name is deprecated since Symfony 3.4 and won\'t be supported in 4.0.'), E_USER_DEPRECATED);
```

Comment:
The message should tell users to override `Kernel::getName()`

Reply:
The idea is to pass the Kernel name as constructor argument, with `app` default for 4.0

Reply:
then it should tell them to pass it in the constructor

Reply:
actually, the deprecation should happen in the constructor itself (when passing `null`)